iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java

```diff
@@ -344,138 +336,161 @@ public void addFile(long size, boolean seq, String name) {
           level,
           count,
           seq ? seqLevelCountGaugeMap : unseqLevelCountGaugeMap,
-          seq ? SEQUENCE : UNSEQUENCE);
-      updateSizeGauge(
+          seq ? SEQUENCE : UNSEQUENCE,
+          FILE_LEVEL_COUNT);
+      updateLevelGauge(
           level,
           totalSize,
           seq ? seqLevelSizeGaugeMap : unseqLevelSizeGaugeMap,
-          seq ? SEQUENCE : UNSEQUENCE);
-      checkIfThereRemainingData();
-    } else {
-      // the metric service has not been set yet
-      hasRemainData.set(true);
+          seq ? SEQUENCE : UNSEQUENCE,
+          FILE_LEVEL_SIZE);
     }
   }
 
-  private void checkIfThereRemainingData() {
-    if (hasRemainData.get()) {
-      synchronized (this) {
-        if (hasRemainData.get()) {
-          hasRemainData.set(false);
-          updateRemainData();
-        }
-      }
-    }
-  }
-
-  private void updateCountGauge(
-      int level, int count, Map<Integer, Gauge> countGaugeMap, String orderStr) {
-    countGaugeMap
+  private void updateLevelGauge(
+      int level, long value, Map<Integer, Gauge> gaugeMap, String orderStr, String gaugeName) {
+    gaugeMap
         .computeIfAbsent(
             level,
             l ->
                 metricService.getOrCreateGauge(
-                    FILE_LEVEL_COUNT,
+                    gaugeName,
                     MetricLevel.CORE,
-                    Tag.TYPE.toString(),
+                    Tag.NAME.toString(),
                     orderStr,
                     LEVEL,
                     String.valueOf(level)))
-        .set(count);
-  }
-
-  private void updateSizeGauge(
-      int level, long size, Map<Integer, Gauge> sizeGaugeMap, String orderStr) {
-    sizeGaugeMap
-        .computeIfAbsent(
-            level,
-            l ->
-                metricService.getOrCreateGauge(
-                    FILE_LEVEL_SIZE,
-                    MetricLevel.CORE,
-                    Tag.TYPE.toString(),
-                    orderStr,
-                    LEVEL,
-                    String.valueOf(level)))
-        .set(size);
+        .set(value);
+  }
+
+  public void deleteRegion(String database, String regionId) {
+    Arrays.asList(seqFileNumMap, unseqFileNumMap)
+        .forEach(map -> deleteRegionFromMap(map, database, regionId));
+    Arrays.asList(seqFileSizeMap, unseqFileSizeMap)
+        .forEach(map -> deleteRegionFromMap(map, database, regionId));
+    Arrays.asList(
+            seqFileNumGaugeMap, unseqFileNumGaugeMap, seqFileSizeGaugeMap, unseqFileSizeGaugeMap)
+        .forEach(map -> deleteRegionFromMap(map, database, regionId));
+  }
+
+  public <T> void deleteRegionFromMap(
+      Map<String, Map<String, T>> map, String database, String regionId) {
+    Map<String, T> innerMap =
+        map.computeIfPresent(
+            database,
+            (k, v) -> {
+              v.remove(regionId);
+              return v;
+            });
+    if (innerMap == null) {
+      map.remove(database);
```

Comment:
Also, atomicity is not guaranteed here, and there may be some concurrency issues. Feel like you need to make a judgment in the function expression and return null if isEmpty?